### PR TITLE
Enable build on ARM(SVE)

### PR DESCRIPTION
**Pull request by ajay-fuji**
# Description
### Enable scikit-learn-intelex build on ARM(SVE) cpu -
Now since oneDAL officially support arm CPUs [PR-2614](https://github.com/oneapi-src/oneDAL/pull/2614), this PR will enable Scikit-learn-intelex build on arm CPUs. 
# Changes proposed in this PR :
  - Remove hardcoded architecture directory (`arch_dir`) name (intel64). Implemented run time selection of `arch_dir` based on backend CPU.

### CODE REVIEW

File: daal4py/__init__.py [hunk 1]
```diff
@@ -21,13 +22,22 @@
...
     import sys
 
+    plt_arch = platform.machine()
+    arch_dir = None
+    if plt_arch == "x86_64":
+        arch_dir = "intel64"
+    elif plt_arch == "aarch64":
+        arch_dir = "arm"
+    else:
+        arch_dir = plt_arch
```
Comment (icfaust) [inline daal4py/__init__.py:32]:
```suggestion
    arch_dir = platform.machine()
    plt_dict = {"x86_64":"intel64", "aarch64":"arm"}
    arch_dir = plt_dict[arch_dir] if arch_dir in plt_dict else arch_dir
```

File: onedal/__init__.py [hunk 2]
```diff
@@ -23,11 +24,20 @@
...
 
+    plt_arch = platform.machine()
+    arch_dir = None
+    if plt_arch == "x86_64":
+        arch_dir = "intel64"
+    elif plt_arch == "aarch64":
+        arch_dir = "arm"
+    else:
+        arch_dir = plt_arch
+
```
Comment (icfaust) [inline onedal/__init__.py:35]:
```suggestion
    arch_dir = platform.machine()
    plt_dict = {"x86_64":"intel64", "aarch64":"arm"}
    arch_dir = plt_dict[arch_dir] if arch_dir in plt_dict else arch_dir
```
More general for possible future expansion.

File: scripts/build_backend.py [hunk 3]
```diff
@@ -167,6 +169,15 @@ def custom_build_cmake_clib(
...
 
+    plt_arch = plt.machine()
+    arch_dir = None
+    if plt_arch == "x86_64":
+        arch_dir = "intel64"
+    elif plt_arch == "aarch64":
+        arch_dir = "arm"
+    else:
+        arch_dir = plt_arch
+
```
Comment (icfaust) [inline scripts/build_backend.py:180]:
```suggestion
    arch_dir = plt.machine()
    plt_dict = {"x86_64":"intel64", "aarch64":"arm"}
    arch_dir = plt_dict[arch_dir] if arch_dir in plt_dict else arch_dir
```

File: setup.py [hunk 4]
```diff
@@ -51,18 +52,27 @@
...
 
+plt_arch = plt.machine()
+arch_dir = None
+if plt_arch == "x86_64":
+    arch_dir = "intel64"
+elif plt_arch == "aarch64":
+    arch_dir = "arm"
+else:
+    arch_dir = plt_arch
+
```
Comment (icfaust) [inline setup.py:63]:
```suggestion
arch_dir = plt.machine()
plt_dict = {"x86_64":"intel64", "aarch64":"arm"}
arch_dir = plt_dict[arch_dir] if arch_dir in plt_dict else arch_dir
```

File: tests/run_examples.py [hunk 5]
```diff
@@ -57,12 +59,21 @@
...
 
+plt_arch = plt.machine()
+arch_dir = None
+if plt_arch == "x86_64":
+    arch_dir = "intel64"
+elif plt_arch == "aarch64":
+    arch_dir = "arm"
+else:
+    arch_dir = plt_arch
+
```
Comment (icfaust) [inline tests/run_examples.py:70]:
```suggestion
arch_dir = plt.machine()
plt_dict = {"x86_64":"intel64", "aarch64":"arm"}
arch_dir = plt_dict[arch_dir] if arch_dir in plt_dict else arch_dir
```

**Review by icfaust — COMMENTED** [review]
Update to generalize beyond just arm

**Comment by ajay-fuji** [comment]
@icfaust are there any recommended changes to be made on this PR?

**Comment by icfaust** [comment]
> @icfaust are there any recommended changes to be made on this PR?

Let me get someone else more experienced to give this PR a look, but it looks fine to me.

**Review by napetrov — APPROVED** [review]

**Review by ethanglaser — CHANGES_REQUESTED** [review]
Please add "AMD64": "intel64" to the plt_dirs

**Review by ethanglaser — APPROVED** [review]

**Pull request merged**